pkg/asset/agent/manifests/clusterimageset.go

```diff
@@ -182,7 +184,9 @@ func (a *ClusterImageSet) validateReleaseVersion(workflowType workflow.AgentWork
 	fieldPath := field.NewPath("Spec", "ReleaseImage")
 
 	releaseImage := &releaseimage.Image{}
-	releaseImage.Generate(asset.Parents{})
+	if err := releaseImage.Generate(ctx, asset.Parents{}); err != nil {
+		panic(err)
```

Comment:
@andfasano maybe we can return an error here instead of panic'ing. WDYT?

Reply:
This is a golint error by the way: it (rightfully?) complained about an unchecked error.

I'm open to ideas. The function doesn't return errors but rather fieldErrors: would an error in generating the `releaseimage` asset be the sign of a field error here? Alternatively, we could change the function signature to also return a plain "failure-type" `error`.

Reply:
Sorry for the late reply! Definitely it'd be better to avoid the panic here. For this specific case I think the `field.InternalError` could be used to report the Generate error

Reply:
Here is what I have force-pushed:

```diff
diff --git a/pkg/asset/agent/manifests/clusterimageset.go b/pkg/asset/agent/manifests/clusterimageset.go
index 28a2cab34b..0822bd18d2 100644
--- a/pkg/asset/agent/manifests/clusterimageset.go
+++ b/pkg/asset/agent/manifests/clusterimageset.go
@@ -185,7 +185,7 @@ func (a *ClusterImageSet) validateReleaseVersion(ctx context.Context, workflowTy

        releaseImage := &releaseimage.Image{}
        if err := releaseImage.Generate(ctx, asset.Parents{}); err != nil {
-               panic(err)
+               return append(allErrs, field.InternalError(fieldPath, fmt.Errorf("failed to generate the release image asset: %w", err)))
        }

        if a.Config.Spec.ReleaseImage != releaseImage.PullSpec {
```

Is this what you meant @andfasano ?

Reply:
Thanks!